compiler/rustc_interface/src/passes.rs

```diff
@@ -385,28 +384,18 @@ pub fn configure_and_expand(
         sess.emit_warning(ProcMacroCratePanicAbort);
     }
 
-    // For backwards compatibility, we don't try to run proc macro injection
-    // if rustdoc is run on a proc macro crate without '--crate-type proc-macro' being
-    // specified. This should only affect users who manually invoke 'rustdoc', as
-    // 'cargo doc' will automatically pass the proper '--crate-type' flags.
-    // However, we do emit a warning, to let such users know that they should
-    // start passing '--crate-type proc-macro'
-    if has_proc_macro_decls && sess.opts.actually_rustdoc && !is_proc_macro_crate {
```

Comment:
I don't have context on this. Why didn't we create a proc macro before? What's the reason for the check you removed?

Reply:
I don't know, but it reads like this has been a warning for 3 years, so let's remove it ^^

Reply:
Please move breaking changes to a different PR.

Reply:
next force push will have it removed from this PR. The new PR is up at https://github.com/rust-lang/rust/pull/107291